### ISSUE TITLE
Only Allow Users to Edit Posts They've Submitted

### ISSUE THREAD

**Issue by GFLEMING133** [msoe-sse/jekyll-post-editor#31]
* Currently users who log onto the site will all have the same visibility for posts. This brings into question of security & visibility.
* Implementing some sort of visibility permissions would likely require a database and hosting.
> AWS, Azure, etc...
* Let's talk about this at the next meeting and decide whether or not this is something we ought to move forward with.

**Comment by GFLEMING133** [comment]
For context, original board content:

----
### Finish Post Editing Phase 3: Only allow users to edit posts they have submitted

Blocked (7/31/19): I think this would be easier with a database and tricky to do with PR API responses. With PRs we first need to fetch all the posts from the _posts folder and the loop through over potentially hundreds of PRs causing performance problems. I think we should reconsider the database idea.

> Full User Story: As a user of the tool, I do not want to be able to edit posts submitted by other users because I don’t want to step on anyone’s toes.

**Comment by andy-wojciechowski** [comment]
@GFLEMING133 I'm playing with 2 ideas I have for this in a sample Ruby on Rails project using PostgresSQL and Airtable here: https://github.com/andy-wojciechowski/postgres-airtable-sample. PostgresSQL is well integrated with Heroku which peaked my interest

In my initial findings with this if we were to use Airtable one of the limitations is querying for data. Which if you look at the sample SSE resources API: https://airtable.com/appnfHuWptUFonq1U/api/docs#curl/introduction you need to pass in the query string in the URL Also, we would be duplicating a lot of code if we make similar requests but they query similar data. I feel like we would get around this information if we use an Airtable Client

As for Airtable Clients the two below are linked specifically in the Airtable API documentation as third-party clients. Both of these are on people's personal GitHub accounts and haven't been touched in a while but the Ruby one looks more built up..
-https://github.com/sirupsen/airrecord - Ruby
-https://github.com/ngocnicholas/airtable.net - .NET

The official API client is written with node: https://github.com/Airtable/airtable.js

Through a simple google search I found a Python client: https://github.com/gtalarico/airtable-python-wrapper. It's not linked in the Airtable documentation but it looks like a lot of work was put into it. 

I'm going to play with the Airrecord client since I'm using Rails in my sample project but if we were to go with Airtable I recommend either Python or Ruby if we were to go third party. Python would be preferable since we can integrate it nicely with the sse-web-api. If we were to go with another language we would need to figure how to fit Airtable into our repos.

**Comment by andy-wojciechowski** [comment]
@GFLEMING133 I finished my small sample application with PostgresSQL here: https://github.com/andy-wojciechowski/postgres-airtable-sample and I deployed the app to heroku here: https://postgres-airtable-sample.herokuapp.com/. The heroku link is using a PostgresSQL hosted by heroku and the development version is using a local version.

**Conclusions**
1. Airtable/PostgresSQL with heroku hosting would cost us nothing as long as we don't upgrade to premium plans
2. Airtable was a little simpler then PostgresSQL to setup and I feel like Airtable would be simpler for Freshman/Sophmores to understand. If we go with PostgresSQL we may need to rely on Juniors/Seniors to manage the database. Also, the code itself might involve a little learning curve for Freshman/Sophmores due to setting up PostgresSQL locally.
3. For airtable we would need to use a client library related to the comment above. My experience using the ruby client Airrecord was good and it was simple to use.
4. For PostgresSQL we can use more mature frameworks, like ActiveRecord in Ruby and for python: https://github.com/psycopg/psycopg2. I used ActiveRecord in the sample application and I felt like it was simple to use since ActiveRecord created most of the methods I needed for me. In total the 1 ActiveRecord class I wrote was only 3 lines of code.
5. If we were to PostgresSQL we would probably need to manage db migrations in order to create the database

I feel like both of these options would be doable. I'd say if we want to keep managing a DB as simple as possible for everyone I'd say Airtable, or if we want to use something a little more built up that we can build into our existing Heroku apps I'd say PostgresSQL.
